collections/ansible_collections/purestorage/flasharray/plugins/modules/purefa_host.py

```diff
@@ -313,23 +320,26 @@ def _is_cbs(array, is_cbs=False):
 
 def _set_host_initiators(module, array):
     """Set host initiators."""
-    if module.params["protocol"] in ["nvme", "mixed"]:
+    if not module.params["protocol"]:
+        module.fail_json(msg="Host creation failed. 'protocol' not specified")
+    if module.params["protocol"].lower() in ["nvme", "mixed"]:
         if module.params["nqn"]:
             try:
                 array.set_host(module.params["name"], nqnlist=module.params["nqn"])
             except Exception:
                 array.delete_host(module.params["name"])
                 module.fail_json(msg="Setting of NVMe NQN failed.")
-    if module.params["protocol"] in ["iscsi", "mixed"]:
+    if module.params["protocol"].lower() in ["iscsi", "mixed"]:
         if module.params["iqn"]:
             try:
                 array.set_host(module.params["name"], iqnlist=module.params["iqn"])
             except Exception:
                 array.delete_host(module.params["name"])
                 module.fail_json(msg="Setting of iSCSI IQN failed.")
-    if module.params["protocol"] in ["fc", "mixed"]:
+    if module.params["protocol"].lower() in ["fc", "mixed"]:
         if module.params["wwns"]:
             try:
+                module.warn("Here")
```

Comment:
Remove this, was probably used for debugging.

Reply:
Done. Thanks for catching that.